File: LongestCommonPrefix.java
import java.util.*;
public class LongestCommonPrefix {
    

    public static String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }

        // Sort the array
        Arrays.sort(strs);

        // Compare the first and last strings in the sorted array
        String first = strs[0];
        String last = strs[strs.length - 1];
        StringBuilder commonPrefix = new StringBuilder();

        // Find the common prefix between the first and last strings
        for (int i = 0; i < Math.min(first.length(), last.length()); i++) {
            if (first.charAt(i) == last.charAt(i)) {
                commonPrefix.append(first.charAt(i));
            } else {
                break;
            }
        }

        return commonPrefix.toString();
    }

    public static void main(String[] args) {
        String[] strs = {"flower", "flow", "flight"};
        System.out.println(longestCommonPrefix(strs));  // Output: "fl"
    }
}

File: LetCode_pactice/FAVQ/LongestCommonPrefix.py
def longest_common_prefix(strs):
    if not strs:
        return ""
    
    # Sort the array
    strs.sort()

    # Compare the first and the last strings
    first = strs[0]
    last = strs[-1]
    common_prefix = ""

    # Find the common prefix between the first and last strings
    for i in range(min(len(first), len(last))):
        if first[i] == last[i]:
            common_prefix += first[i]
        else:
            break

    return common_prefix

# Example usage
strs = ["flower", "flow", "floght"]
print(longest_common_prefix(strs))  # Output: "fl"
